app/workers/publishing_api_legacy_tags_worker.rb

```diff
@@ -0,0 +1,72 @@
+class PublishingApiLegacyTagsWorker
+  include Sidekiq::Worker
+  API = Services.publishing_api
+
+  def perform(model_id, model_class)
+    model = model_class.constantize.find(model_id)
+    previous_version, taxon_ids = get_taxon_ids(model)
+    parent_ids = taxon_ids.flat_map { |id| expand_parents(id) }
+    legacy_ids = get_legacy_links(taxon_ids + parent_ids)
+
+    links = { policy_areas: [], policies: [], topics: [] }
+    update_topics(links, legacy_ids, model)
+    update_policies(links, legacy_ids, model)
+    update_policy_areas(links, legacy_ids, model)
```

Comment:
does the model need persisting? i.e. call model.save

Reply:
None of these methods actually change the model, just relationships on it. So it's not necessary to save the model.